src/dataflow/src/sink/kafka.rs

```diff
@@ -991,16 +1028,6 @@ where
                 }
             }
 
-            let in_flight = s.producer.in_flight_count();
-            s.metrics.messages_in_flight.set(in_flight as u64);
-
-            if !s.ready_rows.is_empty() {
-                // We need timely to reschedule this operator as we have pending
-                // items that we need to send to Kafka
-                s.activator.activate();
-                return true;
-            }
-
             if !s.pending_rows.is_empty() {
```

Comment:
Why do we need this part? Timely should be activating the operator whenever its input frontier advance

Reply:
Does it also do so when the durability frontier advances? (from `source_timestamp_histories`). That seems less clearly codified.

If you know for sure that the answer is yes, I'll remove it.  But, worst case, it feels like us being a little defensive even if it's not strictly necessary